packages/@aws-cdk/cdk/lib/duration.ts

```diff
@@ -0,0 +1,178 @@
+import { Token } from "./token";
+
+/**
+ * Represents a length of time.
+ */
+export class Duration {
+  /**
+   * @param amount the amount of Seconds the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Seconds.
+   */
+  public static seconds(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Seconds);
+  }
+
+  /**
+   * @param amount the amount of Minutes the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Minutes.
+   */
+  public static minutes(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Minutes);
+  }
+
+  /**
+   * @param amount the amount of Hours the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Hours.
+   */
+  public static hours(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Hours);
+  }
+
+  /**
+   * @param amount the amount of Days the `Duration` will represent.
+   * @returns a new `Duration` representing `amount` Days.
+   */
+  public static days(amount: number): Duration {
+    return new Duration(amount, TimeUnit.Days);
+  }
+
+  /**
+   * @param duration an ISO-formtted duration to be parsed.
+   * @returns the parsed `Duration`.
+   */
+  public static parse(duration: string): Duration {
+    const matches = duration.match(/^PT(?:(\d+)D)?(?:(\d+)H)?(?:(\d+)M)?(?:(\d+)S)?$/);
+    if (!matches) {
+      throw new Error(`Not a valid ISO duration: ${duration}`);
+    }
+    const [, days, hours, minutes, seconds] = matches;
+    if (!days && !hours && !minutes && !seconds) {
+      throw new Error(`Not a valid ISO duration: ${duration}`);
+    }
+    return Duration.seconds(
+      _toInt(seconds)
+      + (_toInt(minutes) * 60)
+      + (_toInt(hours) * 3_600)
+      + (_toInt(days) * 86_400)
+    );
+
+    function _toInt(str: string): number {
+      if (!str) { return 0; }
+      return Number(str);
+    }
+  }
+
+  private readonly amount: number;
+  private readonly unit: TimeUnit;
+
+  private constructor(amount: number, unit: TimeUnit) {
+    if (Token.isUnresolved(amount)) {
```

Comment:
Nope... We need to support number tokens as long as they don't require conversion